softlayer/resource_softlayer_global_ip.go

```diff
@@ -0,0 +1,270 @@
+	if err != nil {
+		return fmt.Errorf("Not a valid global ip ID, must be an integer: %s", err)
+	}
+
+	globalIp, err := service.Id(globalIpId).Mask(GlobalIpMask).GetObject()
+	if err != nil {
+		return fmt.Errorf("Error retrieving Global Ip: %s", err)
+	}
+
+	d.Set("id", *globalIp.Id)
+	d.Set("ip_address", *globalIp.IpAddress.IpAddress)
+	if globalIp.DestinationIpAddress != nil {
+		d.Set("routes_to", *globalIp.DestinationIpAddress.IpAddress)
+	}
+	return nil
+}
+
+func resourceSoftLayerGlobalIpUpdate(d *schema.ResourceData, meta interface{}) error {
+	sess := meta.(*session.Session)
+	service := services.GetNetworkSubnetIpAddressGlobalService(sess)
+
+	globalIpId, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return fmt.Errorf("Not a valid global ip ID, must be an integer: %s", err)
+	}
+
+	_, err = service.Id(globalIpId).Route(sl.String(d.Get("routes_to").(string)))
+	if err != nil {
+		return fmt.Errorf("Error editing Global Ip: %s", err)
+	}
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"pending"},
+		Target:  []string{"complete"},
+		Refresh: func() (interface{}, string, error) {
+			transaction, err := service.Id(globalIpId).GetActiveTransaction()
+			if err != nil {
+				return datatypes.Network_Subnet_IpAddress_Global{}, "pending", err
+			}
+			if transaction.Id == nil {
+				return datatypes.Network_Subnet_IpAddress_Global{}, "complete", nil
+			}
+			return datatypes.Network_Subnet_IpAddress_Global{}, "pending", nil
+		},
+		Timeout:    10 * time.Minute,
+		Delay:      5 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	pendingResult, err := stateConf.WaitForState()
+
+	if err != nil {
+		return fmt.Errorf("Error waiting for global ip destination ip address to become active: %s", err)
+	}
+
+	if _, ok := pendingResult.(datatypes.Network_Subnet_IpAddress_Global); ok {
+		return nil
+	}
+
+	return nil
+}
+
+func resourceSoftLayerGlobalIpDelete(d *schema.ResourceData, meta interface{}) error {
+	sess := meta.(*session.Session)
+	service := services.GetNetworkSubnetIpAddressGlobalService(sess)
+
+	globalIpId, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return fmt.Errorf("Not a valid global ip ID, must be an integer: %s", err)
+	}
+
+	billingItem, err := service.Id(globalIpId).GetBillingItem()
+	if err != nil {
+		return fmt.Errorf("Error deleting global ip: %s", err)
+	}
+
+	if billingItem.Id == nil {
+		return nil
+	}
+
+	_, err = services.GetBillingItemService(sess).Id(*billingItem.Id).CancelService()
+
+	return err
+}
+
+func resourceSoftLayerGlobalIpExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	sess := meta.(*session.Session)
+	service := services.GetNetworkSubnetIpAddressGlobalService(sess)
+
+	globalIpId, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return false, fmt.Errorf("Not a valid ID, must be an integer: %s", err)
+	}
+
+	result, err := service.Id(globalIpId).GetObject()
+	if err != nil {
+		return false, fmt.Errorf("Error retrieving global ip: %s", err)
```

Comment:
I think here we need to differentiate between an error with the api, and simply a 404, which would mean it does not exist (we would return `false, nil` in that case).